### ISSUE TITLE
Version 0.12 Potential Items

### ISSUE THREAD

**Issue by davidslater** [twosixlabs/armory#684]
Prioritized by importance and sequence
- [x] Update dependencies & docker images
    - [x] Pin to ART 1.4
    - [x] Add TensorFlow Object Detection API
- [x] Data canonicalization & updating of scenarios to match
- [x] New scenario integration
    - [x] New dataset integration
        - [ ] Dataset upload/integration tool?
    - [x] Baseline model integration
    - [ ] Compression defenses rework?
- [ ] Sanity checking of the model
- [ ] Tagging model output (output format - transcript or prob lattice...; logits vs probs vs log probs)
- [ ] Explicit task / scenario definitions
    - [ ] Use `attack.targeted` as a uniform way of grabbing this from ART to ensure attacks are provided the correct labels. ART 1.4 should implement `"targeted"` as a required property of `EvasionAttack`???
- [ ] Dataset metadata (speaker gender, input shape, sampling rate / frame rate, number of classes)
- [ ] Integrated model training
    - [ ] Method to save models
    - [ ] Single core issues

Potential complications
- [ ] Complex numbers - perturbation metric? Gradients over complex numbers / activation functions?
- [ ] OD output - number of boxes by number of classes; number of boxes by 4 (corners)
- [ ] Audio - differnet performers might be attacking inputs at different spaces (cochleagram)

Next release
- [ ] Improved Jupyter integration
- [ ] Improved attack saving/viewing/iteration
- [ ] Update black formatter version
- [ ] Upgrade to newer docker-py
- [ ] Convenience functions for grabbing models/datasets/etc. in a single line from config file

**Comment by davidslater** [comment]
Unchecked items moved to 0.13
